Content.Server/Chemistry/Metabolism/HealthChangeMetabolism.cs

```diff
@@ -41,14 +41,28 @@ public class HealthChangeMetabolism : IMetabolizable
         /// <param name="solutionEntity"></param>
         /// <param name="reagentId"></param>
         /// <param name="tickTime"></param>
+        /// <param name="availableReagant">Reagant available to be metabolized.</param>
         /// <returns></returns>
-        ReagentUnit IMetabolizable.Metabolize(IEntity solutionEntity, string reagentId, float tickTime)
+        ReagentUnit IMetabolizable.Metabolize(IEntity solutionEntity, string reagentId, float tickTime, ReagentUnit availableReagent)
         {
+            // how much reagant should we metabolize
+            var metabolismAmount = MetabolismRate * tickTime;
+
+            // is that much reagant actually available?
+            if (availableReagent < metabolismAmount) {
+                metabolismAmount = availableReagent;
+            }
+
+            // how much does this much reagant heal for
+            var healthChangeAmmount = HealthChange * metabolismAmount.Float();
```

Comment:
```suggestion
            var healthChangeAmount = HealthChange * metabolismAmount.Float();
```

Reply:
Variable name typo fixed throughout the file with new commit